scripts/create_expression_parser.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""INSTRUCTIONS:
+Run this script from the oppia root folder:
+  python -m scripts.create_expression_parser
+The root folder MUST be named 'oppia'.
+It produces the expression parser.
+"""
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+
+import fileinput
+import os
+import re
+import subprocess
+
+import python_utils
+
+from . import common
+from . import install_third_party_libs
+from . import setup
+
+
+def main():
+    """Produces the expression parser."""
+    setup.main()
+
+    expression_parser_definition = (
+        'core/templates/dev/head/expressions/parser.pegjs')
+    expression_parser_js = (
+        'core/templates/dev/head/expressions/ExpressionParserService.js')
+
+    # Install the basic environment, e.g. nodejs.
+    install_third_party_libs.main([])
+
+    python_utils.PRINT(
+        'Checking whether pegjs is installed in %s' % common.OPPIA_TOOLS_DIR)
+    if not os.path.exists('node_modules/pegjs'):
+        python_utils.PRINT('Installing pegjs')
+        subprocess.call((
+            '%s/bin/npm install pegjs@0.8.0' % common.NODE_PATH).split())
```

Comment:
Can you make a standard utility function in one of the python scripts that handles installation of npm libs, similar to what we did in bash? Have other things that need similar functionality make use of that utility function too.

Ideally you can generalize it to cover the whole block from lines 47-52.

Reply:
Done